test/impl/AzureBlobFiles.test.js

```diff
@@ -571,6 +606,41 @@ describe('_getPresignUrl', () => {
     expect(url).toEqual(expectedUrl)
     expect(tvm.getAzureBlobPresignCredentials).toHaveBeenCalledWith({ blobName: 'fakesub/afile', expiryInSeconds: 60, permissions: 'fake' })
   })
+
+  test('_getPresignUrl with correct options default permission own credentials', async () => {
+    files.hasOwnCredentials = true
+    files.credentials = fakeUserCredentials
+    files._azure.sasCreds = false
+    const cleanUrl = 'https://fake.blob.core.windows.net/fake/fakesub/afile'
+    setMockBlobUrl(cleanUrl)
+    const expectedUrl = 'https://fake.blob.core.windows.net/fake/fakesub/afile?fakeSAS'
+    const url = await files._getPresignUrl('fakesub/afile', { expiryInSeconds: 60 })
+    expect(url).toEqual(expectedUrl)
+  })
+
+  test('_getPresignUrl with correct options explicit permission own credentials', async () => {
+    files.hasOwnCredentials = true
+    files.credentials = fakeUserCredentials
+    files._azure.sasCreds = false
+    const cleanUrl = 'https://fake.blob.core.windows.net/fake/fakesub/afile'
+    setMockBlobUrl(cleanUrl)
+    const expectedUrl = 'https://fake.blob.core.windows.net/fake/fakesub/afile?fakeSAS'
+    const url = await files._getPresignUrl('fakesub/afile', { expiryInSeconds: 60, permissions: 'fake' })
+    expect(url).toEqual(expectedUrl)
+  })
+
+  test('_getAzureBlobPresignCredentials with sas creds', async () => {
+    files.hasOwnCredentials = true
+    await expect(files._getAzureBlobPresignCredentials('fakesub/afile', { expiryInSeconds: 60 })).rejects.toThrow('[FilesLib:ERROR_UNSUPPORTED_OPERATION] Operation generatePresignURL not supported for SAS credentails')
+  })
+
+  test('_getAzureBlobPresignCredentials with account creds', async () => {
+    files.hasOwnCredentials = true
+    files.credentials = fakeUserCredentials
+    files._azure.sasCreds = false
+    const ret = await files._getAzureBlobPresignCredentials('fakesub/afile', { expiryInSeconds: 60 })
+    expect(ret).toEqual({ signature: 'fakeSAS' })
+  })
```

Comment:
missing test with tvm credentials ? you could pass a tvm mock to the init => 
```
mockFn = jest.fn()
mockfiles = await AzureBlobFiles.init(fakeSASCredentials, { getAzureBlobPresignCredentials:  mockFn })
```

then you can test

```
expect(mockFn.getAzureBlobPresignCredentials).toHaveBeenCalled()
```

Reply:
This one is specific unit test for the private function `_getAzureBlobPresignCredentials `, there are other tests which mock tvm

Reply:
Ok I see, I think we should test against `_getPresignUrl'` directly with different tvm, byo account and byo sas credential. I don't think we should test against _getAzureBlobPresignCredentials as it is not exposed by the Files interface